ucsdscheduleme/Repo/ScheduleRepo.cs

```diff
@@ -31,98 +27,105 @@ public class ScheduleRepo
+            for (int i = 0; i < coursesToSchedule.Count() - 1; i++)
             {
-                for (int j = i + 1; j < AllSections.Count(); j++)
+                foreach (Node j in AllSections[i])
                 {
-                    Node node1 = AllSections[i];
-                    Node node2 = AllSections[j];
-                    if (node1.Course != node2.Course)
+                    foreach (Node k in AllSections[i + 1])
                     {
-                        if (!Conflict(node1.Section, node2.Section))
-                        {
-                            node1.Edges.Add(node2);
-                        }
+                        j.Edges.Add(k);
                     }
                 }
             }
 
-            // dfs on graph to find possible schedules
-            foreach (Node section in AllSections)
+            // Call DFS on each subgraph, starting with the first class sections
+            // being the root
+            foreach (Node i in AllSections[0])
             {
-                if (section.IsFirstClass)
-                {
-                    DFS(section, numClasses, numNodes, ref possibleSchedules);
-                }
+                DFS(i, numClasses, ref possibleSchedules);
             }
-
             return possibleSchedules;
         }
 
         /// <summary>
-        /// Performs DFS on schedule graph.
+        /// Performs DFS on schedule graph. Begins with pushing the root onto the stack. Then
+        /// continuously pops and adds the popped section onto the schedule. When a node
+        /// is popped, it will push all its children onto the stack. Once a node from the last
+        /// layer is popped, then this is a unique schedule. This schedule is checked
+        /// for conflicts and then added if there are no time conflicts.
         /// </summary>
         /// <param name="root">Node to do DFS from</param>
         /// <param name="numClasses">Total number of classes to schedule</param>
         /// <param name="possibleSchedules">Modified list of possible schedules</param>
-        private void DFS(Node root, int numClasses, int numNodes, ref List<List<Section>> possibleSchedules)
+        private void DFS(Node section, int numClasses, ref List<List<Section>> possibleSchedules)
         {
-            Node curr;
-            Stack<Node> stack = new Stack<Node>();
-            bool[] isNodeVisited = new bool[numNodes];
-            List<Section> temp = new List<Section>
-            {
-                root.Section
-            };
-            stack.Push(root);
-            isNodeVisited[0] = true;
+            // start stack with the root
+            Stack<Node> s = new Stack<Node>();
+            s.Push(section);
 
-            while (stack.Count != 0)
+            Section[] tempSchedule = new Section[numClasses];
+
+            // Adds the first class in the correct order on the temp schedule, then push children.
+            while (s.Count != 0)
```

Comment:
shouldn't the default condition be 1? Because since the first input is a section in the first class, we assume that we have at least one class. Also, if there is no class, we don't have to compare anything.

Reply:
yeah but eventually we’ll reach the end of the stack :o

Reply:
yeah but eventually we’ll reach the end of the stack :o